apps/aeutil/lib/serialization.ex

```diff
@@ -18,30 +20,59 @@ defmodule Aeutil.Serialization do
 
   @spec block(Block.t() | map(), :serialize | :deserialize) :: map | Block.t()
   def block(block, :serialize) do
-    serialized_block = serialize_value(block)
-    Map.put(serialized_block["header"], "transactions", serialized_block["txs"])
+    serialized_header = serialize_value(block.header)
+    serialized_transactions = Enum.map(block.txs, fn tx -> tx(tx, :serialize) end)
+    %{"header" => serialized_header, "transactions" => serialized_transactions}
   end
 
   def block(block, :deserialize) do
     txs = Enum.map(block["transactions"], fn tx -> tx(tx, :deserialize) end)
 
     built_header =
-      block
-      |> Map.delete("transactions")
+      block["header"]
       |> deserialize_value()
       |> Header.new()
 
     Block.new(header: built_header, txs: txs)
   end
 
-  @spec tx(map(), :deserialize) :: SpendTx.t()
-  def tx(tx, :serialize), do: serialize_value(tx)
+  @spec tx(map(), :serialize | :deserialize) :: SignedTx.t()
+  def tx(tx, :serialize) do
+    case tx.data.payload do
+      %OracleQueryTxData{} ->
+        tx
+        |> serialize_value()
+        |> update_in(["data", "payload", "oracle_address"], &Account.base58c_encode(&1))
+
+      %OracleResponseTxData{} ->
+        tx
+        |> serialize_value()
+        |> update_in(["data", "payload", "query_id"], &OracleQueryTxData.base58c_encode(&1))
+
+      _ ->
+        serialize_value(tx)
+    end
+  end
 
   def tx(tx, :deserialize) do
     tx_data = tx["data"]
+
     data = DataTx.deserialize(tx_data)
+
+    updated_data =
+      case data do
+        %OracleQueryTxData{} ->
+          update_in(data, [:payload, :oracle_address], &Account.base58c_decode(&1))
```

Comment:
Same as the above, but about the docoding func

Reply:
What do you mean?

Reply:
See the below comment. I meesed up the orientation